src/shims/unix/linux/fs.rs

```diff
@@ -0,0 +1,167 @@
+
+                epfd.file_descriptors.remove(&fd);
+                Ok(0)
+            } else {
+                this.handle_not_found()
+            }
+        } else {
+            let einval = this.eval_libc("EINVAL")?;
+            this.set_last_error(einval)?;
+            Ok(-1)
+        }
+    }
+
+    fn epoll_wait(
+        &mut self,
+        epfd: &OpTy<'tcx, Provenance>,
+        events: &OpTy<'tcx, Provenance>,
+        maxevents: &OpTy<'tcx, Provenance>,
+        timeout: &OpTy<'tcx, Provenance>,
+    ) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+
+        let _epfd = this.read_scalar(epfd)?.to_i32()?;
+        let _events = this.read_scalar(events)?;
+        let maxevents = this.read_scalar(maxevents)?.to_i32()?;
+        let _timeout = this.read_scalar(timeout)?.to_i32()?;
+
+        if maxevents <= 0 {
+            throw_ub_format!("maxevents must be greater than 0");
+        }
+
+        throw_ub_format!("closed due to not writing");
+    }
+
+    fn eventfd(
+        &mut self,
+        val: &OpTy<'tcx, Provenance>,
+        flags: &OpTy<'tcx, Provenance>,
+    ) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+
+        let val = this.read_scalar(val)?.to_u32()?;
+        let flags = this.read_scalar(flags)?.to_i32()?;
+
+        let efd_cloexec = this.eval_libc_i32("EFD_CLOEXEC")?;
+        let efd_nonblock = this.eval_libc_i32("EFD_NONBLOCK")?;
+        let efd_semaphore = this.eval_libc_i32("EFD_SEMAPHORE")?;
+
+        if flags & (efd_cloexec | efd_nonblock | efd_semaphore) == 0 {
+            throw_unsup_format!("{flags} is unsupported");
+        }
+        // FIXME handle the cloexec and nonblock flags
+        if flags & efd_cloexec != 0 {}
+        if flags & efd_nonblock != 0 {}
+        if flags & efd_semaphore != 0 {
+            throw_unsup_format!("EFD_SEMAPHORE is unsupported");
+        }
+
+        let fh = &mut this.machine.file_handler;
+        let fd = fh.insert_fd(Box::new(Event { val }));
+        Ok(fd)
+    }
+
+    fn socketpair(
+        &mut self,
+        _domain: &OpTy<'tcx, Provenance>,
+        type_: &OpTy<'tcx, Provenance>,
+        _protocol: &OpTy<'tcx, Provenance>,
+        sv: &OpTy<'tcx, Provenance>,
+    ) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+
+        let _flags = this.read_scalar(type_)?.to_i32()?;
+        let sv = this.deref_operand(sv)?;
+
+        let fh = &mut this.machine.file_handler;
+        let sv0 = fh.insert_fd(Box::new(SocketPair));
+        let sv0 = ScalarInt::try_from_int(sv0, sv.layout.size).unwrap();
+        let sv1 = fh.insert_fd(Box::new(SocketPair));
+        let sv1 = ScalarInt::try_from_int(sv1, sv.layout.size).unwrap();
+
+        this.write_scalar(sv0, &sv.into())?;
+        this.write_scalar(sv1, &sv.offset(sv.layout.size, sv.layout, this)?.into())?;
+
+        Ok(0)
+    }
+
+    fn libc_current_sigrtmax(&mut self) -> InterpResult<'tcx, i32> {
```

Comment:
This does not look like a file system function. Maybe just implement it directly in foreign_items, given how trivial it is?